### PR TITLE
feat: #378 change the filters component

### PR DESCRIPTION
**Pull request by nolliia**
Screenshot 2024-08-22 at 13 49 59

## Summary by Sourcery

Replace the existing filter component with a new implementation using the Disclosure component from Headless UI. Enhance the filter functionality by updating the URL parameters based on selected options, and remove the deprecated MultipleSelector component.

New Features:
- Introduce a new filter component using the Disclosure component from Headless UI, allowing users to expand and collapse filter options.

Enhancements:
- Implement URL parameter updates to reflect selected filter options, enhancing the user experience by maintaining filter state in the URL.

Chores:
- Remove the deprecated MultipleSelector component and its associated imports, streamlining the codebase.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,8 +34,8 @@
     "cryptr": "^6.3.0",
     "date-fns": "^3.3.1",
     "date-fns-tz": "^3.1.3",
-    "isomorphic-dompurify": "^2.14.0",
     "iso-639-3": "^3.0.1",
+    "isomorphic-dompurify": "^2.14.0",
     "jwt-decode": "^4.0.0",
```
Comment (zalborzi) [inline package.json:39]: we have this lib already in our package, I have added it a week ago.

File: package.json [hunk 2]
```diff
@@ -50,12 +51,12 @@
   },
   "devDependencies": {
     "@types/axios": "^0.14.0",
+    "@types/dompurify": "^2.3.0",
```
Comment (zalborzi) [inline package.json:54]: same.

File: package.json [hunk 3]
```diff
@@ -33,8 +34,8 @@
     "cryptr": "^6.3.0",
     "date-fns": "^3.3.1",
     "date-fns-tz": "^3.1.3",
-    "isomorphic-dompurify": "^2.14.0",
     "iso-639-3": "^3.0.1",
+    "isomorphic-dompurify": "^2.14.0",
```
Comment (zalborzi) [inline package.json:38]: we have this lib already in our package, I have added it a week ago.

**Comment by zalborzi** [comment]
I just prefer it if the purple line below the filter name doesn't extend all the way to the box's border. Feel free to ignore this if you don't think it looks nice.

File: package.json (hunk 3 above)
Comment (nolliia) [reply package.json:38]: it was just moved by prettier I didn't change it

File: package.json (hunk 2 above)
Comment (nolliia) [reply package.json:54]: it was just moved by prettier I didn't change it

**Review by zalborzi — APPROVED** [review]

**Pull request merged**